packages/doenetml-worker-rust/doenetml-core/src/utils.rs

```diff
@@ -1,7 +1,27 @@
-// use crate::state::EssentialStateVar;
-// use crate::EssentialDataOrigin;
+use std::collections::HashMap;
 
-use serde_json::{json, Map, Value};
+pub trait KeyValueIgnoreCase<K, V> {
+    fn get_key_value_ignore_case<'a>(&'a self, key: &str) -> Option<(&'a K, &'a V)>;
+}
+
+impl<K, V> KeyValueIgnoreCase<K, V> for HashMap<K, V>
+where
+    K: ToString + std::cmp::Eq + std::hash::Hash,
+{
+    /// Match key to the HashMap keys, ignoring case.
+    /// Return the original key-value pair the HashMap.
+    fn get_key_value_ignore_case<'a>(&'a self, key: &str) -> Option<(&'a K, &'a V)> {
+        let lower_case_key = key.to_lowercase();
+
+        for k in self.keys() {
+            if k.to_string().to_lowercase() == lower_case_key {
```

Comment:
You could even use [`eq_ignore_ascii_case`](https://doc.rust-lang.org/std/primitive.str.html#method.eq_ignore_ascii_case) for slight efficiency gains.

Reply:
Good idea!